### DISCUSSION TITLE
Permissions Missing When Installing As 'Only enable this app' (like Files.Read.All)

### DISCUSSION THREAD

**Discussion by influential-eliot** [microsoft-search/pnp-modern-search#3645] [Ideas]
# Incident
When installing this app in SharePoint Admin for a client it does not request the following permissions:

- Files.Read.All
- Mail.Read
- Presence.Read.All
- Sites.Read.All

## To Reproduce
Detailed steps to reproduce the behavior:
1. On step 2 of the installation steps ([here](https://microsoft-search.github.io/pnp-modern-search/installation/)) select the '_Only enable this app_' option

## Expected behavior
The full set of permissions to be requested, as per the Full Permissions screenshot listed when I added this to a different tenancy using the '_Enable this app and add it to all sites_' option.

Or at the very least the '`Files.Read.All`' permission, otherwise how will it search files on the sites that it is added to?

# Information
## Check existing issues before logging new ones
I have searched all issues for files.read.all and could not obviously see this issue.

## Version used
4.11.0

### Missing Permissions
![missing permissions](https://github.com/microsoft-search/pnp-modern-search/assets/112952958/dae1a16a-f7f9-4b71-a385-ed748cfdb495)

### Full Permissions
![all permissions](https://github.com/microsoft-search/pnp-modern-search/assets/112952958/8b39ce39-bab2-418b-aa3c-cbc113601513)

## Browsers Used
 - FireFox
 - Edge

**Comment by kasperbolarsen** [comment]
To me it sounds like you are asking for those permissions in order to display certain content in the persona card? Have you looked at the options provided when using the LPC option rather than the persona card?
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/2692387c-0e26-4a27-ac26-a49b1fed2f0d)

**Reply by influential-eliot** [reply]
Those options are not present at the *install* stage.

**Reply by influential-eliot** [reply]
So ... yes, perhaps the incident should be re-opened?

**Reply by kasperbolarsen** [reply]
Have you checked if the LPC people card shows the info you are looking for?
If that is the case there os no reason why the app should ask ( optional) for further permissions.

**Reply by influential-eliot** [reply]
I don't know what the LPC people card is.

As far as I have seen on both of my installs on two tenancies, that didn't, and doesn't show when adding the whole thing to the SharePoint tenancy as indicated in the original post.

**Reply by kasperbolarsen** [reply]
The LPC ,Live Person Card, is the out of the box SharePoint people card 😊
I would expect that the pnp-person component we are using now soon will will be deprecated in favor of mgt-people and LPC.
https://microsoft-search.github.io/pnp-modern-search/usage/search-results/layouts/people/

**Reply by influential-eliot** [reply]
Yes ... OK ... and ... when ... **installing this** (_as per my original comments [here](github.com/microsoft-search/pnp-modern-search/issues/3643#discussion_bucket) and [here](https://github.com/microsoft-search/pnp-modern-search/discussions/3645#discussion-6406216)_) ... how is that card relevant to step two?

> On step 2 of the installation steps ([here](https://microsoft-search.github.io/pnp-modern-search/installation/)) select the '_Only enable this app_' option

**Reply by kasperbolarsen** [reply]
You don't have to be granted any of the permissions in order for the LPC to work 😊 None of the permissions are required in order for the PnP Modern Search web parts to work.  In order some of the properties in the People layout/card to show up we would need those permissions,  but that is not the case with LPC. I hope this makes sense.

**Reply by influential-eliot** [reply]
I don't need the LPC to work ... 😅 ... I literally only just learned about the thing because you mentioned it.

(_sorry ... but I cannot stress this enough ... that is not the issue that I am reporting and I am not sure why you are so focused upon it_)

I am reporting that only a subset of permissions are requested when installing and choosing the option labelled '_Only enable this app_' not across the whole tenancy.

**Reply by kasperbolarsen** [reply]
Sorry, I am just so used to people reporting issues where their internal IT would rather die than granting those permissions. 
I re-read your initial post and tested it on a brand new tenant.
As far as I can tell the permissions requested in the App catalog are the same in both "Only enable" and "Enable and add"
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/b6b8c58d-eba8-400c-934a-7616a09918b0)

I checked the API Access settings, but shipped the approval 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/6456dea4-76ae-4f44-8e9d-b43e4e6bf6bc)
 
and tested that the search results web part works just fine as long as the data source is SharePoint

If I switch to using Microsoft Search/Graph I get this
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/ec3e25fd-806d-4dba-8ad7-08fe26b6ac40)

So I guess your issue is only related to Microsoft Search ?

**Reply by influential-eliot** [reply]
When I press through the install (_as per the initial report, in the initial incident_) the only permissions requested, by the app (_not search_) do not include the ones that are supposed to be included.

Is it that some permissions are not viewable by people without certain roles? If so, that is new to me.

Also, when you requested the above, had you canceled the previous install, and removed the app and its permissions as previously requested in the previous attempt?

It seems odd that you are saying quite a lot of stuff that doesn't appear to be connected to what I have done, here ... that's all ... and meanwhile this could be happening again, elsewhere.

**Reply by kasperbolarsen** [reply]
Hi, in order to ensure that my test is as valid as possible,  I created a brand new tenant and added v4.11 to the newly created App Catalog. 
Sorry for asking,  but I have to cover all angles: are you sure that the permissions you are missing hasn't already been granted and thus isn't shown?
Can you verify the issue on a new tenant?

**Reply by influential-eliot** [reply]
I can try it on a third tenancy.

`edit` - Sorry, I am trying, but I am getting the '_Could not retrieve Global Service Principal ID Error: Error: {}_' error in the API permissions ... and [deleting the '_SharePoint Online Client Extensibility Web Application Principal_' Enterprise Application](https://learn.microsoft.com/en-us/answers/questions/1070859/error-could-not-retrieve-global-service-principal) has not assisted in making that error go away. 

OK, magically, on this third tenancy, it is showing those permissions, which no role dependency.

Is there a chance that a tenant can restrict the permissions that a SharePoint App can request, then? As I have shown evidence that this clearly can happen.

If that is indeed possible (the same tenant still only shows the reduced permissions request) then should it perhaps be noted in the install notes that it is possible that not all permissions may be requested?

**Reply by wobba** [reply]
Tenants can mess up the SPFx app used..may be something on sp-dev-docs to assist on getting it to work again.
